cruise-control-core/src/main/java/com/linkedin/cruisecontrol/detector/metricanomaly/PercentileMetricAnomalyFinderConfig.java

```diff
@@ -18,24 +18,51 @@
   public static final String METRIC_ANOMALY_PERCENTILE_UPPER_THRESHOLD_CONFIG =
       "metric.anomaly.percentile.upper.threshold";
   private static final String METRIC_ANOMALY_PERCENTILE_UPPER_THRESHOLD_DOC =
-      "The upper threshold for the metric anomaly detector to identify an increase in the metric "
-          + "values of a broker as a metric anomaly.";
+      "The upper percentile threshold for the metric anomaly detector to identify an increase in the metric "
+      + "values of a broker as a metric anomaly. The current metric value is compared against the historical value "
+      + "corresponding to given percentile in the metric history after the application of the upper margin.";
 
   /**
    * <code>metric.anomaly.percentile.lower.threshold</code>
    */
   public static final String METRIC_ANOMALY_PERCENTILE_LOWER_THRESHOLD_CONFIG =
       "metric.anomaly.percentile.lower.threshold";
   private static final String METRIC_ANOMALY_PERCENTILE_LOWER_THRESHOLD_DOC =
-      "The lower threshold for the metric anomaly detector to identify a decrease in the metric "
-          + "values of a broker as a metric anomaly.";
+      "The lower percentile threshold for the metric anomaly detector to identify a decrease in the metric "
+      + "values of a broker as a metric anomaly. The current metric value is compared against the historical value "
+      + "corresponding to given percentile in the metric history after the application of the lower margin.";
+
+  /**
+   * <code>metric.anomaly.upper.margin</code>
+   */
+  public static final String METRIC_ANOMALY_UPPER_MARGIN_CONFIG = "metric.anomaly.upper.margin";
```

Comment:
The docs becomes hard to understand after adding the margin values. I think it'll be useful to give an example and have the docs reference the example.